src/main/python/tests/examples/tutorials/test_adult.py

```diff
@@ -0,0 +1,243 @@
+        Yt = self.sds.from_numpy(test_labels[:test_count])
+        Yt = Yt + 1.0
+
+        betas = multiLogReg(X, Y)
+
+        [_, y_pred, acc] = multiLogRegPredict(Xt, betas, Yt).compute()
+
+        self.assertGreater(acc, 80)
+        
+        confusion_matrix_abs, _ = confusionMatrix(self.sds.from_numpy(y_pred), Yt).compute()
+
+        self.assertTrue(
+            np.allclose(
+                confusion_matrix_abs,
+                np.array([[3583,  502],
+                         [245,  670]])
+            )
+        )
+
+
+    def test_neural_net(self):
+        # Reduced because we want the tests to finish a bit faster.
+        train_count = 15000
+        test_count = 5000
+
+        train_data, train_labels, test_data, test_labels = self.d.get_preprocessed_dataset(interpolate=True, standardize=True, dimred=0.1)
+
+        # Train data
+        X = self.sds.from_numpy( train_data[:train_count])
+        Y = self.sds.from_numpy( train_labels[:train_count])
+
+        # Test data
+        Xt = self.sds.from_numpy(test_data[:test_count])
+        Yt = self.sds.from_numpy(test_labels[:test_count])
+
+        FFN_package = self.sds.source(self.neural_net_src_path, "fnn", print_imported_methods=True)
+
+        network = FFN_package.train(X, Y, 1, 16, 0.01, 1)
+
+        self.assertTrue(type(network) is not None) # sourcing and training seems to works
+
+        FFN_package.save_model(network, '"model/python_FFN/"').compute(verbose=True)
+
+        # TODO This does not work yet, not sure what the problem is
+        #probs = FFN_package.predict(Xt, network).compute(True)
+        # FFN_package.eval(Yt, Yt).compute()
+
+
+    def test_parse_dataset_with_systemdsread(self):
+        # Reduced because we want the tests to finish a bit faster.
+        train_count = 30000
+        test_count = 10000
+        #self.sds.read(self.dataset_path_train, schema=self.dataset_path_train_mtd).compute(verbose=True)
+        print("")
+
+        SCHEMA = '"DOUBLE,STRING,DOUBLE,STRING,DOUBLE,STRING,STRING,STRING,STRING,STRING,DOUBLE,DOUBLE,DOUBLE,STRING,STRING"'
+        F1 = self.sds.read(
+            self.dataset_path_train,
+            schema=SCHEMA
+        )
+        F2 = self.sds.read(
+            self.dataset_path_test,
+            schema=SCHEMA
+        )
+        jspec = self.sds.read(self.dataset_jspec, data_type="scalar", value_type="string")
+        #scaling does not have effect yet. We need to replace labels in test set with the same string as in train dataset
+        X1, M1 = F1.rbind(F2).transform_encode(spec=jspec).compute()
+        col_length = len(X1[0])
+        X = X1[0:train_count, 0:col_length-1]
+        Y = X1[0:train_count, col_length-1:col_length].flatten()-1
+        # Test data
+        Xt = X1[train_count:train_count+test_count, 0:col_length-1]
+        Yt = X1[train_count:train_count+test_count, col_length-1:col_length].flatten()-1
+
+        _ , mean , sigma = scale(self.sds.from_numpy(X), True, True).compute()
```

Comment:
here also we should not need to compute, but simply get the values out, and call apply on the test data.

The first result from this call is the scaled and shifted X, therefore the following second scaling of x is not needed.